blog/tidy-collaborative-filtering/index.qmd

```diff
@@ -0,0 +1,214 @@
+
+## Content-Based Filering
+
+Content-based filtering methods exploit information about users or items to create recommendations by building a model that relates available characteristics of users or items to each other. The recommendation problem is hence cast into a classification problem (user will like the item or not) or more generally a regression problem (which rating will user give an item). The classification problem can be item-centered by focusing on available user information and estimating a model for each item. If there are a lot of user-item interactions available, the resulting model is fairly robust, but it is less personalized (as it ignores user characteristics apart from interactions). The classification problem can also be user-centered by working with item features and estimating a model for each user. However, if a user only has few interactions then the resulting model becomes easily instable. Content-based filtering can also be neither user nor item-centered by stacking the two feature vectors and putting them into a neural network. 
+
+The main advantage of content-based filtering is that it can make recommendations for new users without any interaction history or recommend new items to users. The disadvantages include: (i) training needs a lot of users and item examples for reliable results; (ii) tuning might be much harder in practice than collaborative filtering; (iii) missing information might be a problem since there is no clear solution how to treat missingness in user or item characteristics.
+
+## Hybrid Recommenders
+
+Hybrid recommender systems combine both collaborative and content-based filtering to overcome the challenges of each individual approach. There are different hybridization techniques available, e.g., combining the scores of different components (weighted), choosing among different compontent (switching), following strict priority rules (cascading), presenting outputs from different components at the same time (mixed), etc.
+
+# Train Collaborative Filtering Recommenders in R
+
+For this post, we rely on the `tidyverse` [@Wickham2019] family of packages, `scales` [@scales] for scale functions for visualization, and `recommenderlab` [@recommenderlab] - a package that provides an infrastructure to develop and test collaborative filerting recommender algorithms.  
+```{r, message = FALSE, warning = FALSE}
+library(tidyverse)
+library(scales)
+library(recommenderlab)
+```
+
+```{r, include = FALSE}
+wikifolio_portfolios <- read_rds("../data/wikifolio_portfolios.rds")
+```
+
+I load a dataset with stock holdings of investable wikifolios at the beginning of 2023. Reach out if you want to get access to the data. 
+
+```{r}
+glimpse(wikifolio_portfolios)
+```
+
+First, I convert the long data to a binary rating matrix where only non-NA values are stored explicitly and NA values are represented by dots for efficiency. 
```

Comment:
I don't believe that NAs are represented by dots is solving anything. Under the hood, much more is happening to store sparse matrices with the arules package.

Reply:
I don't remember what I wanted to say here when I wrote this. Removed "where only non-NA values are stored explicitly and NA values are represented by dots for efficiency"